### DISCUSSION TITLE
How to configure two gRPC ports to serve different services?

### DISCUSSION THREAD

**Discussion by sunng87** [quarkusio/quarkus#30251] [Q&A] [answered]
In our application, there are two sets of gRPC services for both internet and intranet. I'd like to configure serve them with two different port and add TLS configurations.

I checked the doc and it doesn't seem to have the configuration. So is it supported or configurable?

[a comment by an automated account is not shown]

**Comment by alesj** [comment, marked as answer]
> So is it supported or configurable?

Currently no, afaik.
As this changes things a lot ... e.g. which services, interceptors belong to which server, etc.

You'll need 2 diff apps / microservices ...

**Comment by sunng87** [comment]
That's unfortunate. But please consider this feature. It can be like `clients` to have its own name.

**Comment by drsarutobi8** [comment]
I might be possible with combination of envoy proxy as a sidecar. Assuming that all internet services are also available to intranet users.
 
Service A port 9000 -> envoy proxy -> 9090 
Service B 9000

Internet user: port 9090 Service A (GRPC, REST)
Intranet users: port 9000 Service A,B (GRPC)

I am working on project with gRPC JSON transcoding (to REST) wtih envoy and found that port working with both REST and GRPC so this might be a workaround for you.
